### ISSUE TITLE
Sitespeed and Quantum(?)

### ISSUE THREAD

**Issue by rschoessler** [sitespeedio/sitespeed.io#1785]
So I downloaded Firefox Quantum beta in anticipation of next week's release.  I wanted to do a sitespeed comparison of pages in Chrome and Firefox.   When I ran in Firefox, I got the below error.  I'll admit that I don't typically run performance tests in Firefox with Sitespeed so I'm not sure if I'd get this error with another version.  So I thought I would post this in case it's something you are not aware of.

Running version sitespeed.io 6.0.0-alpha1; Firefox 57.0b14 (64-bit)

```
[2017-11-08 09:06:43] ERROR: WebDriverError
    at Object.throwDecodedError (c:\sitespeed\sitespeed.io\node_modules\selenium-webdriver\lib\error.js:497:15)
    at parseHttpResponse (c:\sitespeed\sitespeed.io\node_modules\selenium-webdriver\lib\http.js:519:13)
    at doSend.then.response (c:\sitespeed\sitespeed.io\node_modules\selenium-webdriver\lib\http.js:441:30)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
    at Driver.schedule (c:\sitespeed\sitespeed.io\node_modules\selenium-webdriver\lib\webdriver.js:807:17)
    at Navigation.to (c:\sitespeed\sitespeed.io\node_modules\selenium-webdriver\lib\webdriver.js:1131:25)
    at Driver.get (c:\sitespeed\sitespeed.io\node_modules\selenium-webdriver\lib\webdriver.js:988:28)
    at context.runWithDriver (c:\sitespeed\sitespeed.io\bin\prescript.js:5:21)
    at SeleniumRunner.runWithDriver (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\seleniumRunner.js:217:12)
    at Object.runWithDriver (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\engine.js:324:31)
    at Object.run (c:\sitespeed\sitespeed.io\bin\prescript.js:4:20)
    at Promise.mapSeries.preScript (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\engine.js:315:23)
From previous event:
    at Promise.resolve.tap.tap (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\engine.js:314:19)
From previous event:
    at runIteration (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\engine.js:313:10)
    at Promise.reduce (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\engine.js:409:27)
From previous event:
    at Promise.resolve.tap.tap.tap.tap.result (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\engine.js:406:17)
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
    at processImmediate [as _immediateCallback] (timers.js:714:5)
From previous event:
    at Engine.run (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\engine.js:405:8)
    at engine.start.then (c:\sitespeed\sitespeed.io\lib\plugins\browsertime\analyzer.js:95:26)
    at Server.once (events.js:342:8)
    at Server.listen (net.js:1422:10)
    at Object.startServer (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\support\extensionServer.js:15:19)
    at Engine.startExtensionServer (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\engine.js:145:8)
    at Engine.start (c:\sitespeed\sitespeed.io\node_modules\browsertime\lib\core\engine.js:175:12)
    at Object.analyzeUrl (c:\sitespeed\sitespeed.io\lib\plugins\browsertime\analyzer.js:94:8)
From previous event:
    at Object.analyzeUrl (c:\sitespeed\sitespeed.io\lib\plugins\browsertime\analyzer.js:95:8)
    at storageManager.createDirForUrl.then.then (c:\sitespeed\sitespeed.io\lib\plugins\browsertime\index.js:173:32)
    at c:\sitespeed\sitespeed.io\node_modules\mkdirp\index.js:30:20
    at FSReqWrap.oncomplete (fs.js:135:15)
From previous event:
    at Object.processMessage (c:\sitespeed\sitespeed.io\lib\plugins\browsertime\index.js:173:12)
    at queue.process.message (c:\sitespeed\sitespeed.io\lib\support\queueHandler.js:192:32)
    at drainItem (c:\sitespeed\sitespeed.io\node_modules\concurrent-queue\index.js:96:21)
    at Immediate.drain (c:\sitespeed\sitespeed.io\node_modules\concurrent-queue\index.js:77:84)
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
    at processImmediate [as _immediateCallback] (timers.js:714:5)
```

**Comment by soulgalore** [comment]
Hi @rschoessler short summary: Quantum (beta/nightly whatever) should work as long as you turn off collecting the HAR (**--browsertime.skipHar**). Firefox 55 broke the https://github.com/firebug/har-export-trigger/ that we use to collect the HAR. I've been asking for a while that there will be an version for 57 and they (Mozilla) has it on the 2do list for this quarter. The plan for us was to release 6.0 asap when Quantum was released, but the HAR will still be broken, so lets see how we do it. If you run our Docker container we still use 54 there.

It looks from your log that there maybe is something going on with the WebDriver (and we maybe need to update Geckodriver, I can check that later). How do you start your run (I see that you are in Windows)?

I tried out Browsertime directly and it works on my Mac:
`bin/browsertime.js --firefox.nightly --skipHar https://www.sitespeed.io `

Best
Peter

**Comment by rschoessler** [comment]
I start my run from command line on my windows machine after doing a `git pull` so that I have the latest & greatest.  My command looks similar to:

```c:\sitespeed\sitespeed.io\bin>node sitespeed.js --preScript prescript.js https://page.to.test.com -n 5 -b firefox ```

**Comment by soulgalore** [comment]
@rschoessler we got 57 running in Docker for https://dashboard.sitespeed.io/dashboard/db/browsers where it works, can you try again with the stable 57 and include --skipHar in your run?

**Comment by rschoessler** [comment]
**TL;DR:**  Status looks to be the same/similar to before.  However, maybe this is related to Firefox and preScript??

..........
Here's my current status:

Created a new clone of sitespeed from github
Run a simple test works fine:
`c:\sitespeed2\sitespeed.io\bin>node sitespeed.js https://my.testsite.com/ -n 1 -b firefox --browsertime.firefox.acceptInsecureCerts --browsertime.skipHar`

Run a test with preScript in *Chrome* works fine:
`c:\sitespeed2\sitespeed.io\bin>node sitespeed.js --preScript prescript.js https://my.testsite.com/ -n 1 -b chrome`

**Run a test with *preScript* in *Firefox* fails:**
`c:\sitespeed2\sitespeed.io\bin>node sitespeed.js --preScript prescript.js  https://my.testsite.com/ -n 1 -b firefox --browsertime.firefox.acceptInsecureCerts --browsertime.skipHar`

Error:

> [2017-11-27 10:11:14] INFO: Versions OS: win32 10.0.10586 nodejs: v8.4.0 sitespeed.io: 6.0.0 browsertime: 2.0.0 coach: 1.1.0
> [2017-11-27 10:11:15] INFO: Starting firefox for analysing https://my.testsite.com/home 1 time(s)
> [2017-11-27 10:11:15] INFO: Testing url https://my.testsite.com/home run 1
> [2017-11-27 10:11:19] ERROR: UnknownCommandError: POST /session/9eba953f-4056-49a3-a3be-70e0a6ddcdd6/element/515628e0-db22-40ce-963f-9ea0c0640d5b/submit did not match a known command
>     at Object.throwDecodedError (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\error.js:514:15)
>     at parseHttpResponse (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\http.js:519:13)
>     at doSend.then.response (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\http.js:441:30)
>     at <anonymous>
>     at process._tickCallback (internal/process/next_tick.js:188:7)
>     at Driver.schedule (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\webdriver.js:807:17)
>     at WebElementPromise.schedule_ (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\webdriver.js:2010:25)
>     at WebElementPromise.submit (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\webdriver.js:2338:17)
>     at driver.get.then (c:\sitespeed2\sitespeed.io\bin\prescript.js:15:21)
>     at ManagedPromise.invokeCallback_ (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\promise.js:1376:14)
>     at TaskQueue.execute_ (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\promise.js:3084:14)
>     at TaskQueue.executeNext_ (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\promise.js:3067:27)
>     at asyncRun (c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\promise.js:2927:27)
>     at c:\sitespeed2\sitespeed.io\node_modules\selenium-webdriver\lib\promise.js:668:7
>     at <anonymous>
>     at process._tickCallback (internal/process/next_tick.js:188:7)
> From previous event:
>     at Promise.resolve.tap.tap (c:\sitespeed2\sitespeed.io\node_modules\browsertime\lib\core\engine.js:231:19)
> From previous event:
>     at runIteration (c:\sitespeed2\sitespeed.io\node_modules\browsertime\lib\core\engine.js:230:10)
>     at Promise.reduce (c:\sitespeed2\sitespeed.io\node_modules\browsertime\lib\core\engine.js:330:27)
> From previous event:
>     at Promise.resolve.tap.tap.tap.tap.result (c:\sitespeed2\sitespeed.io\node_modules\browsertime\lib\core\engine.js:327:17)
>     at runCallback (timers.js:781:20)
>     at tryOnImmediate (timers.js:743:5)
>     at processImmediate [as _immediateCallback] (timers.js:714:5)
> From previous event:
>     at Engine.run (c:\sitespeed2\sitespeed.io\node_modules\browsertime\lib\core\engine.js:326:8)
>     at engine.start.then (c:\sitespeed2\sitespeed.io\lib\plugins\browsertime\analyzer.js:114:26)
>     at Server.once (events.js:342:8)
>     at Server.listen (net.js:1422:10)
>     at Object.startServer (c:\sitespeed2\sitespeed.io\node_modules\browsertime\lib\support\extensionServer.js:15:19)
>     at Engine.startExtensionServer (c:\sitespeed2\sitespeed.io\node_modules\browsertime\lib\core\engine.js:54:8)
>     at Engine.start (c:\sitespeed2\sitespeed.io\node_modules\browsertime\lib\core\engine.js:87:12)
>     at Object.analyzeUrl (c:\sitespeed2\sitespeed.io\lib\plugins\browsertime\analyzer.js:113:8)
> From previous event:
>     at Object.analyzeUrl (c:\sitespeed2\sitespeed.io\lib\plugins\browsertime\analyzer.js:114:8)
>     at storageManager.createDirForUrl.then.then (c:\sitespeed2\sitespeed.io\lib\plugins\browsertime\index.js:182:22)
>     at c:\sitespeed2\sitespeed.io\node_modules\mkdirp\index.js:30:20
>     at FSReqWrap.oncomplete (fs.js:135:15)
> From previous event:
>     at Object.processMessage (c:\sitespeed2\sitespeed.io\lib\plugins\browsertime\index.js:181:12)
>     at queue.process.message (c:\sitespeed2\sitespeed.io\lib\core\queueHandler.js:210:32)
>     at drainItem (c:\sitespeed2\sitespeed.io\node_modules\concurrent-queue\index.js:96:21)
>     at Immediate.drain (c:\sitespeed2\sitespeed.io\node_modules\concurrent-queue\index.js:77:84)
>     at runCallback (timers.js:781:20)
>     at tryOnImmediate (timers.js:743:5)
>     at processImmediate [as _immediateCallback] (timers.js:714:5)
> [2017-11-27 10:11:19] INFO: HTML stored in c:\sitespeed2\sitespeed.io\bin\sitespeed-result\my.testsite.com\2017-11-27-10-11-14
> [2017-11-27 10:11:19] INFO: Finished analysing https://my.testsite.com/home

**Comment by soulgalore** [comment]
@rschoessler ah thanks. yes that error you get is from Selenium, I know there's been issues in the past with Firefox (our default test login script on Wikipedia didn't work) but I thought that would be fixed now when we upgraded Geckodriver.

`UnknownCommandError: POST /session/9eba953f-4056-49a3-a3be-70e0a6ddcdd6/element/515628e0-db22-40ce-963f-9ea0c0640d5b/submit did not match a known command`

Seems that that Selenium command don't work with the combination of Firefox/Geckodriver, can you give us just that line with the POST? Or can you see anything when you check the latest docs: http://seleniumhq.github.io/selenium/docs/api/javascript/index.html

**Comment by rschoessler** [comment]
@soulgalore there's actually no line in my prescript that's doing a `POST` per se.  My script is a simple login script.  It dies after it loads the page and doesn't actually enter any data.  (works fine in Chrome).

```'use strict';
module.exports = {
  run(context) {
    return context.runWithDriver((driver) => {
      return driver.get('https://my.testsite.com/login')
        .then(() => {
          var webdriver = context.webdriver;		
          var userName = 'username';
          var password = 'password';
          var loginForm = driver.findElement(webdriver.By.id('login'));
          var loginInput = driver.findElement(webdriver.By.id('UserName'));
          loginInput.sendKeys(userName);
          var passwordInput = driver.findElement(webdriver.By.id('Password'));
          passwordInput.sendKeys(password);
          loginForm.submit();
        });
    })
  }
};```

...not sure if I answered what you were asking.

**Comment by soulgalore** [comment]
Thanks @rschoessler yep the POST is the Webdriver API. I'll test again later this week.

**Comment by mgasiorowski** [comment]
This is bug in selenium:
https://github.com/SeleniumHQ/selenium/issues/4359

You can click to submit button, this works on chrome and firefox. For example:
```
const loginButton = driver.findElement(By.id('login-button'));
loginButton.click();
```

**Comment by soulgalore** [comment]
@mgasiorowski ah thanks, let me update the example the coming days. Best Peter

**Comment by soulgalore** [comment]
This should be fixed now.  Use FF 60 or 61.
